src/System.Management.Automation/security/SecurityManager.cs

```diff
@@ -535,15 +535,15 @@ private static Signature GetSignatureWithEncodingRetry(string path, ExternalScri
             // try harder to validate the signature by being explicit about encoding
             // and providing the script contents
             string verificationContents = Encoding.Unicode.GetString(script.OriginalEncoding.GetPreamble()) + script.ScriptContents;
-            signature = SignatureHelper.GetSignature(path, verificationContents);
+            signature = SignatureHelper.GetSignature(path, Encoding.Unicode.GetBytes(verificationContents));
```

Comment:
The existing code seems wrong here -- the `OriginalEncoding` could be an encoding other than `Unicode`, and that means `Encoding.Unicode.GetBytes(verificationContents)` could be different byte sequences than the original bytes from the file.
I will make a change and see if all tests still pass.